File: android/ui-framework-sample/ui-framework/src/main/java/kh/mcnc/lib/components/BizSortSheet.kt
package kh.mcnc.lib.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

@Composable
fun BizSortSheet(
    isVisible: Boolean,
    selectedSort: String,
    onDismissRequest: () -> Unit,
    onChange: (String) -> Unit
) {
    val options = listOf(
        "distance" to "By Distance",
        "name" to "By Name"
    )

    BizBottomSheet(
        isVisible = isVisible,
        onDismissRequest = onDismissRequest
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 24.dp)
                .padding(bottom = 32.dp)
        ) {
            Text(
                text = "Sort",
                fontSize = 20.sp,
                fontWeight = FontWeight.Bold,
                color = Color(0xFF1A2A5E),
                modifier = Modifier.padding(bottom = 8.dp)
            )
            Text(
                text = "More convenient to locate the branch you wish to visit.",
                fontSize = 14.sp,
                color = Color(0xFF555555),
                modifier = Modifier.padding(bottom = 24.dp)
            )

            Column(
                modifier = Modifier.fillMaxWidth(),
                verticalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                options.forEach { (value, label) ->
                    val isActive = selectedSort == value
                    Surface(
                        shape = RoundedCornerShape(12.dp),
                        color = Color.White,
                        border = BorderStroke(
                            width = if (isActive) 2.dp else 1.dp,
                            color = if (isActive) Color(0xFF1A2A5E) else Color(0xFFEAEAEA)
                        ),
                        modifier = Modifier
                            .fillMaxWidth()
                            .clickable {
                                onChange(value)
                                CoroutineScope(Dispatchers.Main).launch {
                                    delay(200)
                                    onDismissRequest()
                                }
                            }
                    ) {
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(horizontal = 20.dp, vertical = if (isActive) 15.dp else 16.dp),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Text(
                                text = label,
                                fontSize = 15.sp,
                                color = Color(0xFF111111),
                                fontWeight = if (isActive) FontWeight.Bold else FontWeight.Normal
                            )

                            Box(
                                modifier = Modifier
                                    .size(22.dp)
                                    .background(Color.Transparent, CircleShape)
                                    .padding(if (isActive) 0.dp else 0.5.dp), // Adjust for border width visually
                                contentAlignment = Alignment.Center
                            ) {
                                Surface(
                                    modifier = Modifier.fillMaxSize(),
                                    shape = CircleShape,
                                    color = Color.Transparent,
                                    border = BorderStroke(
                                        width = if (isActive) 2.5.dp else 1.5.dp,
                                        color = if (isActive) Color(0xFF1A2A5E) else Color(0xFFEAEAEA)
                                    )
                                ) {}
                                if (isActive) {
                                    Box(
                                        modifier = Modifier
                                            .size(10.dp)
                                            .background(Color(0xFF1A2A5E), CircleShape)
                                    )
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

File: ios/ui-framework-sample/UIFramework/Sources/UIFramework/BizSortSheet.swift
import SwiftUI

public struct BizSortSheetModifier: ViewModifier {
    @Binding var isPresented: Bool
    @Binding var selectedSort: String
    var onChange: ((String) -> Void)?
    
    let options = [
        ("distance", "By Distance"),
        ("name", "By Name")
    ]
    
    public func body(content: Content) -> some View {
        content.bizBottomSheet(
            isPresented: $isPresented,
            content: {
                VStack(alignment: .leading, spacing: 0) {
                    Text("Sort")
                        .font(.system(size: 20, weight: .bold))
                        .foregroundColor(Color(hex: "#1A2A5E"))
                        .padding(.bottom, 8)
                    
                    Text("More convenient to locate the branch you wish to visit.")
                        .font(.system(size: 14))
                        .foregroundColor(Color(hex: "#555555"))
                        .padding(.bottom, 24)
                    
                    VStack(spacing: 16) {
                        ForEach(options, id: \.0) { option in
                            SortOptionRow(
                                label: option.1,
                                isSelected: selectedSort == option.0,
                                action: {
                                    selectedSort = option.0
                                    onChange?(option.0)
                                    // Auto close after small delay
                                    DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
                                        isPresented = false
                                    }
                                }
                            )
                        }
                    }
                }
                .padding(.horizontal, 24)
                .padding(.top, 8)
                .padding(.bottom, 32)
            }
        )
    }
}

private struct SortOptionRow: View {
    let label: String
    let isSelected: Bool
    let action: () -> Void
    
    var body: some View {
        Button(action: action) {
            HStack {
                Text(label)
                    .font(.system(size: 15, weight: isSelected ? .bold : .regular))
                    .foregroundColor(Color(hex: "#111111"))
                
                Spacer()
                
                ZStack {
                    Circle()
                        .stroke(isSelected ? Color(hex: "#1A2A5E") : Color(hex: "#EAEAEA"), lineWidth: isSelected ? 2.5 : 1.5)
                        .frame(width: 22, height: 22)
                    
                    if isSelected {
                        Circle()
                            .fill(Color(hex: "#1A2A5E"))
                            .frame(width: 10, height: 10)
                    }
                }
            }
            .padding(.horizontal, 20)
            .padding(.vertical, isSelected ? 15 : 16)
            .background(Color.white)
            .cornerRadius(12)
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(isSelected ? Color(hex: "#1A2A5E") : Color(hex: "#EAEAEA"), lineWidth: isSelected ? 2 : 1)
            )
        }
    }
}

public extension View {
    func bizSortSheet(
        isPresented: Binding<Bool>,
        selectedSort: Binding<String>,
        onChange: ((String) -> Void)? = nil
    ) -> some View {
        self.modifier(BizSortSheetModifier(
            isPresented: isPresented,
            selectedSort: selectedSort,
            onChange: onChange
        ))
    }
}
